src/runner/kafka.ts

```diff
@@ -192,31 +183,48 @@ class KafkaRunner extends BaseRunner
     }
   };
 
+  /**
+   * Important to note that Kafka is different to SQS and Redis Runners
+   * as with they are both polling consumers
+   * no timeout behaviour, need to hook into the stream provided by node-rdkafka
+   */
   process(topics: Array<string>) {
     return new Promise<KafkaConsumer>((resolve, reject) => {
       const timeoutId = setTimeout(() => {
         this.logger.error('Connection timed out');
         reject();
-      }, (this.config as KafkaConfiguration).connectionTimeout!);
+      }, this.config.connectionTimeout);
+
+      setInterval(() => {
+        if (this.paused) {
+          this.consumer.pause(this.consumer.position());
```

Comment:
Not entirely sure what these are, they don't seem to be documented in the node-rdkafka readme 🤔 

With kafka we could consider just disconnecting on pause / reconnecting as the bookkeeping is done kafka-side. There is obviously a little more to it with re-assignment of partitions etc however (which the client will be handling for us anyway)